### PR TITLE
Feature/sample migrations

### PR DESCRIPTION
**Pull request by JDziewonski98**
# Summary

Some features not directly related to an issue.
1. fixed the create mural controller to properly parse for longitude and latitude to create a coordinate.
2. added a new way to run the app: "npm run start-migrate" will start the app with some premade data in the database,
as seen in models/associationsetup.ts sampleMigrations(). Should be useful for dev purposes.

## Test Plan

Manually with postman

## Related Issues
none

### CODE REVIEW

File: backend/src/services/mural.service.ts [hunk 1]
```diff
@@ -3,7 +3,16 @@ import { UpdateOptions } from "sequelize";
 
 export class MuralService {
   public async create(mural: MuralInterface) {
-    const createdMural: Mural = await Mural.create<Mural>(mural);
+    const coordinates = {
```
Comment (JDziewonski98) [inline backend/src/services/mural.service.ts:6]: this part is weird maybe, the intent is to have just longitude and latitude passed from client for simplicity, but on the databse side we have to package them into an object as so when we create a mural... anyone know a better way?

File: backend/src/services/mural.service.ts [hunk 2]
```diff
@@ -3,7 +3,16 @@ import { UpdateOptions } from "sequelize";
...
+      coordinates: {
+        type: "Point",
+        coordinates: [mural.longitude, mural.latitude],
+      },
+    };
+    const createdMural: Mural = await Mural.create<Mural>({
+      ...mural,
+      ...coordinates,
```
Comment (JDziewonski98) [inline backend/src/services/mural.service.ts:14]: the ... lets us pass a new object that has the attributes of mural and of coordinates, so essentially 'mural' with an added attribute:
 coordinates: {
        type: "Point",
        coordinates: [mural.longitude, mural.latitude],
      }

File: backend/src/models/associationsetup.ts [hunk 3]
```diff
@@ -7,33 +7,87 @@ import { database } from "../config/database";
...
     await database.sync({ force: true });
+    if (process.env.MIGRATE == "true") {
+      await this.sampleMigrations();
+    }
+  }
+
+  // Creates some initial data in database for dev purposes.
+  private async sampleMigrations() {
+    console.log("Creating some data...");
+    const artistNames = ["Michaelangelo", "Banksy"];
```
Comment (tedspare) [inline backend/src/models/associationsetup.ts:45]: love to see it

File: backend/src/models/associationsetup.ts [hunk 4]
```diff
@@ -7,33 +7,87 @@ import { database } from "../config/database";
```
Comment (tedspare) [inline backend/src/models/associationsetup.ts:7]: is associationsetup meant to be run as a test/demo?

File: backend/package.json [hunk 5]
```diff
@@ -6,6 +6,7 @@
   "scripts": {
     "test": "cross-env NODE_ENV=test jest --runInBand",
     "start": "cross-env NODE_ENV=dev nodemon ./src/server.ts",
+    "start-migrate": "cross-env NODE_ENV=dev MIGRATE=true nodemon ./src/server.ts",
```
Comment (tedspare) [inline backend/package.json:9]: is this meant to be run once (and the command will later be removed) or every time?

Comment (tedspare) [reply backend/package.json:9]: in the first case, migration would be an interesting topic for the next team meeting. it definitely wasn't covered in COMP303 but comes up in real-world applications

Comment (JDziewonski98) [reply backend/package.json:9]: so its just an alternate command you can use to run the application for dev purposes. "npm start" will clear the database when it connects, "npm run start-migrate" will create some sample data in the database when it starts up, just for dev / testing purposes, so you dont have to manually create some artists, boroughs, murals, etc every time you want to try the api. If you close the app then run "npm start", that data will be gone and you will have an empty database. If you think it is better, we can make it such that "npm start" does not clear your personal database, and as such your data persists between runs of the backend, but then if you want to clear the database you have to connect to the container yourself.

Comment (JDziewonski98) [reply backend/package.json:9]: Now we will have to work on migrating the existing data from airtable they gave us to a production database at some point, but that has not been worked on yet. how soon do you think we should work on setting up deployment / production environment? asap?

File: backend/src/models/associationsetup.ts (hunk 4 above)
Comment (JDziewonski98) [reply backend/src/models/associationsetup.ts:7]: it is run once whenever the app is started, it is where the relationships between models are defined through sequelize and the ORM syncs up with the database.

Comment (JDziewonski98) [reply backend/src/models/associationsetup.ts:7]: Yes, the relationships are also defined in the column definitions of the models in each particular model class, but for whatever reason sequelize also wants you to define them like this.

File: backend/package.json (hunk 5 above)
Comment (tedspare) [reply backend/package.json:9]: Late response! The alternative start script with mock data for dev makes sense.

As for migration, Lea gave us access to the Airtable. Soph suggested a one-time Python script or similar to do the migration. We're in no rush for production data though - the next priority will be having basic full-stack functionality.

File: backend/src/models/associationsetup.ts (hunk 4 above)
Comment (tedspare) [reply backend/src/models/associationsetup.ts:7]: Sequelize was only helpful up until now lol

**Review by tedspare — APPROVED** [review]
LGTM

**Pull request merged**